pagefaultgames/pokerogue
# [Bug] Run History function saves the most-recent session data possible.

## What are the changes the user will see?
Changes that occur during the last wave will be accurately recorded in the SessionSaveData that goes into Run History at the end of the run. This includes item usage, leveling, evolution, mid-battle move learning, etc. Previously, the wave as-is would be saved.

## Why am I making these changes?
My silly feature.

## What are the changes from a developer perspective?
gameOverPhase now has a new function, identical to game-data.ts' getSessionSaveData(), that creates a SessionSaveData object that accurately represents the wave's events. Because game-data.ts' getSessionSaveData() is a private function, I felt that it was best to create a new private function within gameOverPhase since creating SessionSaveData should be a limited functionality. 

### Screenshots/Videos
Release Pokemon in party --> Lose --> Run History

https://github.com/user-attachments/assets/fa50ef5a-5a6f-4bd7-bdf4-96784ddd321f

## How to test the changes?
Any changes that occur during the last wave of a run should appear in Run History.

## Checklist
